elmslie-samples/android-compose/src/main/java/vivid/money/elmslie/samples/android/compose/paging/view/PagingScreen.kt

```diff
@@ -0,0 +1,137 @@
+package vivid.money.elmslie.samples.android.compose.paging.view
+
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.material.*
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import com.google.accompanist.swiperefresh.SwipeRefresh
+import com.google.accompanist.swiperefresh.rememberSwipeRefreshState
+import vivid.money.elmslie.samples.android.compose.core.EffectWithKey
+import vivid.money.elmslie.samples.android.compose.paging.elm.PagingEffect
+import vivid.money.elmslie.samples.android.compose.paging.elm.PagingState
+
+@Composable
+fun PagingScreen(
+    state: PagingState,
+    effect: EffectWithKey<PagingEffect>?,
+    onRefresh: () -> Unit,
+    onReloadScreen: () -> Unit,
+    onReloadPage: () -> Unit,
+    onCloseToEnd: () -> Unit,
+) {
+    val scaffoldState = rememberScaffoldState()
+    Scaffold(scaffoldState = scaffoldState) {
+        when {
+            state.error != null && state.items == null -> ErrorState(onReloadScreen)
+            state.error == null && state.items == null -> Shimmers()
+            state.items != null -> List(state, onRefresh, onCloseToEnd, onReloadPage)
+        }
+        effect?.takeIfInstanceOf<PagingEffect.RefreshError>()?.key?.let {
```

Comment:
seems a bit overcomplicated, why don't use it like `effect: PagingEffect?`

Reply:
i suggest to move the discussion to our private repo, so that everyone could join it

Reply:
I've tried to describe the reasoning behing it in the javadoc in the vivid money mr